src/probes/journal.c

```diff
@@ -151,17 +161,11 @@ static bool process_existing_entries(sd_journal *journal)
         ret = read_new_entries(journal);
         if (ret < 0) {
                 return false;
-        }
-
-        if (!payload) {
+        } else if (ret == 0) {
```

Comment:
So, I think that I want to continue ignoring `send_data` failures for now, since this may indicate a configuration problem with the telemetry client on the system. Also, I think it's acceptable for the journal probe to be "lossy" with respect to the data it collects.

But I will revisit the error handling here later.